website/content/docs/configuration/controller.mdx

```diff
@@ -61,15 +61,23 @@ description will be read.
 - `auth_token_time_to_live` - Maximum time to live (TTL) for all auth tokens globally (pertains
   to all tokens from all auth methods). Valid time units are anything specified by Golang's
   [ParseDuration()](https://golang.org/pkg/time/#ParseDuration) method. Default is 7 days.
+
 - `auth_token_time_to_stale` - Maximum time of inactivity for all auth tokens globally (pertains
   to all tokens from all auth methods). Valid time units are anything specified by Golang's
   [ParseDuration()](https://golang.org/pkg/time/#ParseDuration) method. Default is 1 day.
 
+- `health_grace_period` - Amount of time Boundary will wait before initiating the shutdown procedure,
```

Comment:
This may already have been widely discussed and decided, if so, ignore me, but my initial impression upon reading this configuration variable name was that it had something to do with how often the health value would update or something. What do you think about sticking a `shutdown` somewhere into this variable? `shutdown_health_grace_period`?

Reply:
This var is actually controlling how long Boundary will wait before shutdown - having it as `shutdown_health_grace_period` or `health_shutdown_grace_period` seems a bit misleading. At least for me, it leads to thinking about the particular health endpoint and how this var might control something _within_ that endpoint.

Maybe we could have this just be `shutdown_grace_period`. Considering that this var is exposed within a `controller` block and it'll affect the controller. This still seems misleading though - this var will actually prevent _any_ Boundary component from shutdown on the entire instance. Perhaps the best thing to do here is call it `shutdown_grace_period` but have it defined in top level context? I'm not sure.

WDYT?

Reply:
I see, yeah it risks being confusing no matter how you slice it I think - this is essentially a `wait_this_long_until_actually_shutting_down` right? What is the important part to convey to the user when they configure this? Why do they want to configure it?

Reply:
> this is essentially a wait_this_long_until_actually_shutting_down right?


Exactly. The reason a user might want to configure this is explained in the listeners documentation:

>This is designed to allow an operator to configure load-balancers to preemptively stop new traffic to a Boundary instance that is going away.

Reply:
I agree with a lot of this discussion, I am not sold on any of the options mentioned so far. Some additional options:

- `graceful_shutdown_period`
- `graceful_shutdown_wait_duration`

Reply:
I think one of the considerations we should make is that this variable is related to the health endpoint. We could mention it just in the docs blurb but it feels strange to me because I see it as a crucial bit of information - If the health endpoint didn't start, we won't use it, even if it's configured... so I'd think the name ought to allude to health _somehow_

Reply:
Changed it to `graceful_shutdown_wait_duration` in [70560f6](https://github.com/hashicorp/boundary/pull/1882/commits/70560f63799e99433406728b0a2c7fa91f6768b8)